### PR TITLE
Add sample_cohort variable #570

### LINKED ISSUE #570: Add documentation for sample cohorts
Also add `sample_cohort` as a variable.

Reported here: https://github.com/pystatgen/sgkit/issues/566#issue-892815600

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #570. This just adds the missing variable, since we already have some documentation about sample cohorts: https://pystatgen.github.io/sgkit/latest/getting_started.html#cohorts

### CODE REVIEW

File: sgkit/variables.py [hunk 1]
```diff
@@ -451,6 +451,15 @@ def _check_field(
     )
 )
 
+sample_cohort, sample_cohort_spec = SgkitVariables.register_variable(
+    ArrayLikeSpec(
+        "sample_cohort",
+        ndim=1,
+        kind="i",
+        __doc__="""The index of the cohort that each sample belongs to.""",
```
Comment (timothymillar) [inline sgkit/variables.py:459]: Minor suggestion: mention that a negative value indicates a sample is not a member of any cohort.

**Review by timothymillar — APPROVED** [review]
LGTM

**Review by jeromekelleher — APPROVED** [review]
LGTM

File: sgkit/variables.py (hunk 1 above)
Comment (tomwhite) [reply sgkit/variables.py:459]: Good point - thanks!

**Pull request merged**